webapp/src/ts/app.component.ts

```diff
@@ -643,4 +651,20 @@ export class AppComponent implements OnInit {
     // avoid Failed to fetch errors being logged when the browser window is reloaded
     this.changesService.killWatchers();
   }
+
+  private async initAnalyticsModules() {
+    try {
+      const canAccess= await this.headerTabsService.canAccessTab('analytics');
+
+      if (!canAccess) {
+        return;
+      }
+
+      const modules = await this.analyticsModulesService.get();
+      this.analyticsActions.setAnalyticsModules(modules || []);
```

Comment:
I see we have this fallback to an empty array set in multiple places. Why not have it in just one place: the reducer. 
We could initialize the value with an empty array (in the default state), and when the value is set, the reducer can cast and fallback.

Reply:
Fallback removed.